rasa/graph_components/validators/config_validator.py

```diff
@@ -0,0 +1,532 @@
+        has_regex_extractor = (
+            RegexEntityExtractorGraphComponent in self._component_types
+        )
+
+        regex_entity_types = {rf["name"] for rf in training_data.regex_features}
+        overlap_between_types = training_data.entities.intersection(regex_entity_types)
+        has_overlap = len(overlap_between_types) > 0
+
+        if has_general_extractors and has_regex_extractor and has_overlap:
+            rasa.shared.utils.io.raise_warning(
+                f"You have an overlap between the "
+                f"{RegexEntityExtractorGraphComponent.__name__} and the "
+                f"statistical entity extractors "
+                f"{types_to_str(present_general_extractors)} "
+                f"in your pipeline. Specifically both types of extractors will "
+                f"attempt to extract entities of the types "
+                f"{', '.join(overlap_between_types)}. "
+                f"This can lead to multiple "
+                f"extraction of entities. Please read "
+                f"{RegexEntityExtractorGraphComponent.__name__}'s "
+                f"documentation section to make sure you understand the "
+                f"implications: {DOCS_URL_COMPONENTS}#regexentityextractor"
+            )
+
+    def _raise_if_featurizers_are_not_compatible(self,) -> None:
+        """Raises or warns if there are problems regarding the featurizers.
+
+        Raises:
+            `InvalidConfigException`
+        """
+        featurizers: List[SchemaNode] = [
+            node
+            for node in self._graph_schema.nodes.values()
+            if issubclass(node.uses, Featurizer2)
+        ]
+        Featurizer2.raise_if_featurizer_configs_are_not_compatible(
+            [schema_node.config for schema_node in featurizers]
+        )
+
+    def _validate_core(self, story_graph: StoryGraph, domain: Domain) -> None:
+        """Validates whether the configuration matches the training data.
+
+        Args:
+           story_graph: a story graph (core training data)
+           domain: the domain
+        """
+        if not self._policy_schema_nodes and story_graph.story_steps:
+            rasa.shared.utils.io.raise_warning(
+                "Found data for training policies but no policy was configured.",
+                docs=DOCS_URL_POLICIES,
+            )
+        if not self._policy_schema_nodes:
+            return
+        self._warn_if_no_rule_policy_is_contained()
+        self._raise_if_domain_contains_form_names_but_no_rule_policy_given(domain)
+        self._raise_if_a_rule_policy_is_incompatible_with_domain(domain)
+        self._warn_if_priorities_are_not_unique()
+        self._warn_if_rule_based_data_is_unused_or_missing(story_graph=story_graph)
+
+    def _warn_if_no_rule_policy_is_contained(self) -> None:
+        """Warns if there is no rule policy among the given policies.
+
+        NOTE: core/policies/_ensemble/check_for_important_policies
```

Comment:
Can we drop the notes as they'll be invalid once we drop the old things?

Reply:
yep sure - this was just for making coding / reviewing easier